lib/IDE/CodeCompletionResultBuilder.h

```diff
@@ -372,13 +372,8 @@ class CodeCompletionResultBuilder {
 
     PrintOptions PO;
     PO.SkipAttributes = true;
-    std::string TypeName;
-    if (IsIUO) {
-      assert(Ty->getOptionalObjectType());
-      TypeName = Ty->getOptionalObjectType()->getStringAsComponent(PO) + "!";
```

Comment:
In theory, this shouldn't crash. But this crash was caused by
https://github.com/apple/swift/blob/e9cb62d476496787ccc92c506e9059db58c96614/lib/IDE/CodeCompletion.cpp#L2177-L2180

`expectedNumParams != BodyParams->size()` is not enough.

In fact, this fix still doesn't produce correct result, but at least, it fixes the crashes. I want to include this fix for 4.2. I will fix this `FIXME` for later releases.